airbyte-integrations/connectors/source-mysql-v2/src/main/kotlin/io/airbyte/integrations/source/mysql/cdc/MySqlDebeziumOperations.kt

```diff
@@ -84,6 +95,121 @@ class MySqlDebeziumOperations(
+            return abortCdcSync()
+        }
+        val savedStateOffset: SavedOffset = parseSavedOffset(debeziumInput)
+
+        val savedGtidSet = MySqlGtidSet(savedStateOffset.gtidSet)
+        val availableGtidSet = MySqlGtidSet(gtidSet)
+        if (!savedGtidSet.isContainedWithin(availableGtidSet)) {
+            log.info {
+                "Connector last known GTIDs are $savedGtidSet, but MySQL server only has $availableGtidSet"
+            }
+            return abortCdcSync()
+        }
+
+        // newGtidSet is gtids from server that hasn't been seen by this connector yet. If the set
+        // exists, check that they are not purged, or we may lose those data.
+        val newGtidSet = availableGtidSet.subtract(savedGtidSet)
+        if (!newGtidSet.isEmpty) {
+            val purgedGtidSet = queryPurgedIds()
+            if (newGtidSet.subtract(purgedGtidSet).equals(newGtidSet)) {
+                log.info {
+                    "Connector has not seen GTIDs $newGtidSet, but MySQL server has purged $purgedGtidSet"
+                }
+                return abortCdcSync()
+            }
+        }
+        val existingLogFiles: List<String> = getBinaryLogFileNames()
+        val found = existingLogFiles.contains(savedStateOffset.position.fileName)
+        if (!found) {
+            log.info {
+                "Connector last known binlog file ${savedStateOffset.position.fileName} is not found in the server"
+            }
+            return abortCdcSync()
+        }
+        return CdcStateValidateResult.VALID
+    }
+    private fun abortCdcSync(): CdcStateValidateResult {
+        if (
+            (configuration.cursorConfiguration as CdcCursor)
+                .invalidCdcBehavior
+                .asInvalidCdcPositionBehavior() is FailSync
+        ) {
+            log.info { "Current position is invalid. Aborting sync." }
+            return CdcStateValidateResult.INVALID_ABORT
+        } else {
+            return CdcStateValidateResult.INVALID_RESET
+        }
+    }
+
+    private fun parseSavedOffset(debeziumInput: DebeziumInput): SavedOffset {
+        val position: MySqlPosition = position(debeziumInput.state.offset)
+        val gtidSet: String? = debeziumInput.state.offset.wrapped.values.first()["gtids"]?.asText()
+        return SavedOffset(position, gtidSet)
+    }
+
+    private fun getFileOffsetBackingStore(
+        properties: Map<String, String>
+    ): FileOffsetBackingStore? {
+        val fileOffsetBackingStore = KafkaConnectUtil.fileOffsetBackingStore()
+        val propertiesMap = properties.toMutableMap()
+        propertiesMap[WorkerConfig.KEY_CONVERTER_CLASS_CONFIG] = JsonConverter::class.java.name
+        propertiesMap[WorkerConfig.VALUE_CONVERTER_CLASS_CONFIG] = JsonConverter::class.java.name
+        fileOffsetBackingStore.configure(StandaloneConfig(propertiesMap))
+        fileOffsetBackingStore.start()
+        return fileOffsetBackingStore
+    }
+
+    private fun getOffsetStorageReader(
```

Comment:
this function is not called anywhere
you can also get rid of keyConverter and valueConverter
this then extends to INTERNAL_CONVERTER_CONFIG and CONNECTOR_NAME_PROPERTY